### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by Shaheryar0054**
- Created a new app. Give it a meaningful name.
- Initialize your project with Git.
- Build my project schema.
- Created and run the necessary migration files.
- Table and column names should match the ERD diagram.
- note: photo for users table should be a link to an image
- Foreign keys should be included.
- Pay attention that in posts table the column holding the foreign key to the users table should be named author_id
- All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

**Review by shella12 — COMMENTED** [review]
Hi @Shaheryar0054 , 

#  Status: Invalidated :red_circle: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have professional README

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20230817151734_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.integer :comments_counter, default: 0
+      t.integer :likes_counter, default: 0
+      t.references :users, null: false, foreign_key: true
```
Comment (rloterh) [inline db/migrate/20230817151734_create_posts.rb]: - [ ] `Author` is aliasing `User` model and thus we have `AuthorId` used as reference for associated models to `User`, as seen in the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Hence, establishing this relationship in the migration query should be similar to:

```ruby
...
  def change
    add_reference :posts, :author, references: :users, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :posts do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

File: db/migrate/20230817152126_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.references :users, null: false, foreign_key: true
```
Comment (rloterh) [inline db/migrate/20230817152126_create_comments.rb]: - [ ] Similarly`Author` is aliasing `User` model and thus we have `AuthorId` used as reference for associated models to `User`, as seen in the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Hence, establishing this relationship in the migration query should be similar to:

```ruby
...
  def change
    add_reference :comments, :author, references: :users, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :comments do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

File: db/migrate/20230817152938_create_likes.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :users, null: false, foreign_key: true
```
Comment (rloterh) [inline db/migrate/20230817152938_create_likes.rb]: - [ ] `Author` is aliasing `User` model and thus we have `AuthorId` used as reference for associated models to `User`, as seen in the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Hence, establishing this relationship in the migration query should be similar to:

```ruby
...
  def change
    add_reference :posts, :author, references: likes, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :likes do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Shaheryar, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🚦 __🏃 🏡 🏃‍♂️ 🏚️ 🌴 🗼

## Highlights 👍
- Good job so far creating migration query!
- ReadMe looks good!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @Shaheryar0054 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good job implementing changes
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**